core/src/main/java/org/elasticsearch/action/support/broadcast/node/TransportBroadcastByNodeAction.java

```diff
@@ -384,7 +384,11 @@ private void onShardOperation(final NodeRequest request, final Object[] shardRes
                 e.setIndex(shardRouting.getIndex());
                 e.setShard(shardRouting.shardId());
                 shardResults[shardIndex] = e;
-                logger.debug("[{}] failed to execute operation for shard [{}]", e, actionName, shardRouting.shortSummary());
+                if (logger.isDebugEnabled() && !TransportActions.isShardNotAvailableException(e)) {
```

Comment:
> it is to me, buy hey, taste :) up to you.

Okay, this made me consider it again and now I agree. I pushed 69fa5bc2367435f1939eaf307eb3be736ebd9740.